src/main/java/net/openhft/chronicle/wire/WireMarshaller.java

```diff
@@ -1622,6 +1699,22 @@ protected void setValue(Object o, ValueIn read, boolean overwrite) {
             throw new UnsupportedOperationException();
         }
 
+        @Override
+        protected void setDefaultValue(Object ignored, Object o) {
+            // TODO very limited form of deep reset, check for immutability
+            if (defaultValue != null && defaultValue.isEmpty()) {
+                Collection coll = unsafeGetObject(o, offset);
+                if (coll == null) {
+                    coll = collectionSupplier.get();
+                    unsafePutObject(o, offset, coll);
+                }
+                coll.clear();
+                return;
+            }
+
+            unsafePutObject(o, offset, defaultValue);
+        }
```

Comment:
This seems pretty nasty, especially given collections are usually mutable, how does it compare to existing behaviour?

Reply:
Yes, that's what the default implementation of `copy()` does, which is called in this method by default.

Reply:
I've removed `setDefaultValue` for collections/maps since they were not doing anything that `copy()` won't do anyway.